File: Java/src/main/java/Solution101_200/Solution171.java
package Solution101_200;

/**
 * @author windows
 */
public class Solution171 {
    public int titleToNumber(String columnTitle) {
        int sum = 0;
        for (int i = 0; i < columnTitle.length(); i++) {
            sum *= 26;
            sum += columnTitle.charAt(i) - 'A' + 1;
        }
        return sum;
    }


    public static void main(String[] args) {
        Solution171 solution171 = new Solution171();
    }
}

File: C++/src/Solution101_200/Solution171.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution171.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution171::titleToNumber(string columnTitle) {
    int result = 0;
    for (char c : columnTitle) {
        result = result * 26 + (c- 'A' + 1);
    }
    return result;
}
